witness/golang/internal/witness/db_test.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package witness
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"testing"
+
+	_ "github.com/mattn/go-sqlite3" // Load drivers for sqlite3
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// TestRoundTrip tests the happy paths (adding stuff and looking it up).
+func TestRoundTrip(t *testing.T) {
+	for _, test := range []struct {
+		desc      string
+		logID     string
+		c         Chkpt
+		extraRuns int
+	}{
+		{
+			desc:  "simple",
+			logID: "monkeys",
+			c: Chkpt{
+				Size: 123,
+				Raw:  []byte("bananas"),
+			},
+		}, {
+			desc:  "check no failure on inserting same data twice",
+			logID: "monkeys",
+			c: Chkpt{
+				Size: 123,
+				Raw:  []byte("bananas"),
+			},
+			extraRuns: 1,
+		},
+	} {
+		t.Run(test.desc, func(t *testing.T) {
+			ctx := context.Background()
+			db, err := sql.Open("sqlite3", ":memory:")
+			if err != nil {
+				t.Error("failed to open temporary in-memory DB", err)
```

Comment:
This should probably be `Fatalf`. Usually the "setup phase" errors, like this one, mean that the rest of the test won't work, so the test should terminate.

Reply:
Okay, changed this here and in a few other places.